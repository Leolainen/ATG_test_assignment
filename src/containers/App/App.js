import React from "react";
import PropTypes from "prop-types";
import classnames from "classnames";
import styles from "./style.module.css";
import Container from "components/Container";

const App = (props) => {
  const { className, children, ...other } = props;

  return (
    <div className={classnames(styles.root, className)} {...other}>
      <header className={styles.header}>This is header</header>

      <Container component="main" className={styles.main}>
        {children}
      </Container>
    </div>
  );
};

App.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
};

App.uiName = "App";

export default App;
