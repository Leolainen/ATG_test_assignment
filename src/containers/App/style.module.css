.root {
  min-height: 100vh;
  max-width: 100vw;
}

.header {
  padding: 16px 32px;
  background-color: var(--harmony-3);
  box-shadow: 0 2.8px 2.2px rgba(197, 188, 232, 0.07),
    0 6.7px 5.3px rgba(197, 188, 232, 0.05),
    0 12.5px 10px rgba(197, 188, 232, 0.042),
    0 22.3px 17.9px rgba(197, 188, 232, 0.035),
    0 41.8px 33.4px rgba(197, 188, 232, 0.028),
    0 100px 80px rgba(197, 188, 232, 0.02);
}

.main {
  margin-top: 64px;
}
