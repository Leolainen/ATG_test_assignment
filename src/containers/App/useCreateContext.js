import { useState, useCallback } from "react";

export const defaultState = {
  game: [],
  products: [],
  searchResults: {},
  isLoading: false,
};

export default (initialState = defaultState) => {
  const initial = initialState;
  const [searchResults, setSearchResults] = useState(initial.searchResults);
  const [products, setProducts] = useState(initial.products);
  const [game, setGame] = useState(initial.game);
  const [isLoading, setIsLoading] = useState(initial.isLoading);

  const onFetchStart = useCallback(() => {
    console.log("onFetchStart");
    setIsLoading(true);
  }, []);

  const onFetchEnd = useCallback(() => {
    console.log("onFetchEnd");
    setIsLoading(false);
  }, []);

  return {
    game,
    setGame,
    products,
    setProducts,
    isLoading,
    searchResults,
    setSearchResults,
    onFetchStart,
    onFetchEnd,
  };
};
