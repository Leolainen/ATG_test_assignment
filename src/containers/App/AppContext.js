import React from "react";
import PropTypes from "prop-types";
import useCreateContext from "./useCreateContext";

export const AppContext = React.createContext({});

export function AppContextProvider(props) {
  const context = useCreateContext(props);
  return (
    <AppContext.Provider value={context}>{props.children}</AppContext.Provider>
  );
}

export const useAppContext = () => React.useContext(AppContext);

AppContextProvider.propTypes = {
  children: PropTypes.node.isRequired,
};

export default AppContext;
