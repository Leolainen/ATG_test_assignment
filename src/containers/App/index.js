import React from "react";
import PropTypes from "prop-types";
import App from "./App";
import { AppContextProvider } from "./AppContext";

const EnhancedApp = (props) => {
  const { children, ...other } = props;

  return (
    <AppContextProvider {...other}>
      <App>{children}</App>
    </AppContextProvider>
  );
};

EnhancedApp.propTypes = {
  children: PropTypes.node.isRequired,
};

export default EnhancedApp;
