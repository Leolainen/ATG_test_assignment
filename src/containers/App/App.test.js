import React from "react";
import { render } from "@testing-library/react";
import App from "./App";

describe("<App />", () => {
  describe("Snapshots", () => {
    it("should render", () => {
      const { asFragment } = render(<App>App</App>);
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
