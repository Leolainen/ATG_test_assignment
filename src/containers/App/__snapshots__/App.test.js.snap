// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Snapshots should render 1`] = `
<DocumentFragment>
  <div
    class="root"
  >
    <header
      class="header"
    >
      This is header
    </header>
    <main
      class="root maxWidth main"
    >
      App
    </main>
  </div>
</DocumentFragment>
`;
